kafka: update the kafka to 2.3.0 (#50)
kudobuilder/frameworks: repository/kafka/docs/latest/configuration.md

# Configuration 

### Resources

> By default, KUDO Kafka resource configuration is set to be minimal optimal for the production usage. 

Reviewer: ```suggestion
By default, KUDO Kafka resource configuration is set to the minimum recommended values for production usage. 
```

> But users can tune the requirements depending on high or low workload of the Kafka cluster.  

Reviewer: ```suggestion
But users can and should tune the configurations based on the workload requirements of their Kafka cluster.  
```

##### Tuning the resources for the Kafka Cluster

```

> kubectl kudo install kafka --instance=my-kafka-name \

Reviewer: Are users able to run this multiple times with different settings and Kudo will automatically know to adjust the current deployment? Or is this only valid for initial deployment?

Author: custom settings are saved in the `instance` object. So once installed you cannot re-install but you can update the settings. [update docs](https://github.com/kudobuilder/operators/blob/d0a944961fedaf199a13814cd1ea92e1168273cc/repository/kafka/docs/latest/update.md) have information on how to do

Reviewer: Right, what I was getting at was that a user would have to run `patch`, not `install`. So they can't really "tune" the cluster this way. I'll suggest a change in the header.

            -p ZOOKEEPER_URI=zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181 \
            -p ZOOKEEPER_PATH=/custom-path \

>             -p BROKER_CPUS=3000m

Reviewer: None of these additional params need `\`?

Author: let me fix that!

            -p BROKER_COUNT=5
            -p BROKER_MEM=4096m
            -p DISK_SIZE=20Gi
            -p MIN_INSYNC_REPLICAS=3
            -p NUM_IO_THREADS=10
            -p NUM_NETWORK_THREADS=5
          
```

##### Storage

By default, the Kafka operator will use the default storage class of the Kubernetes cluster. 

> To deploy the Kafka cluster using a storage class that isn't the default, you can use the parameter `STORAGE_CLASS`

Reviewer: ```suggestion
To deploy Kafka using a different storage class, you can use the parameter `STORAGE_CLASS`
```

```
kubectl kudo install kafka --instance=my-kafka-name -p STORAGE_CLASS=<STORAGE_CLASS_NAME>
```

##### Deploying an ephemeral Kafka cluster without persistence

```
kubectl kudo install kafka --instance=my-kafka-name -p PERSISTENT_STORAGE=false
```

> Having `PERSISTENT_STORAGE` value `false` means that any data or logs in broker will be lost after the a pod restart or rescheduling.

Reviewer: ```suggestion
Having `PERSISTENT_STORAGE` value `false` means that any data or logs inside the brokers will be lost after a pod restart or rescheduling.
```

Reviewer: Should this setting even be allowed? Sounds like a place for future trouble

Author: If there is no `storageclass` in a Kubernetes cluster. Users cannot play with `KUDO Kafka`

Deploying without persistent storage isn't recommended for production usage. 

##### Metrics

> By default, the Kafka cluster will have the JMX Exporter enabled. You can check more information around how KUDO Kafka exposes metrics in [monitoring](./monitoring.md) documentation.

Reviewer: ```suggestion
By default, the Kafka cluster will have the JMX Exporter enabled. You can check more information around how KUDO Kafka exposes metrics in [monitoring](./monitoring.md).
```

##### Zookeeper PATH

> By default, the Kafka cluster will try to use a zookeeper path same as the instance name.

Reviewer: This is a little confusing. What path will it try to use by default? `/zookeeper` ?

Author: The Kafka instance name. By default `KUDO` generates an instance name. But users can specify using `--instance` flag.

Reviewer 2: maybe we could state that explicitly, that it refers to whatever you provided in `--instance` or what was generated for you, which you can figure out by `kubectl get instances`

Author: I have added now both cases. Which pretty much explain what would be the `Zookeeper Path` in case we set the parameter and when we don't set the parameter

> User can override the default zk path by using `ZOOKEEPER_PATH` parameter.

Reviewer: ```suggestion
To override the default zk path, use the `ZOOKEEPER_PATH` parameter.
```

```

> kubectl kudo install kafka --instance=my-kafka-name -p ZOOKEEPER_URI=zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181 -p ZOOKEEPER_PATH=/custom-path

Reviewer: ```suggestion
kubectl kudo install kafka --instance=my-kafka-name \
  -p ZOOKEEPER_URI=zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181 \
  -p ZOOKEEPER_PATH=/custom-path
```
Easier to identify the `ZOOKEEPER_PATH`

```

##### Docker image

> The [Dockerfile](https://github.com/mesosphere/dcos-kafka-service/blob/master/images/Dockerfile) used in the KUDO Kafka operator is hosted in the [dcos-kafka-service](https://github.com/mesosphere/dcos-kafka-service) repo

Reviewer: ```suggestion
The Dockerfile used to build the KUDO Kafka operator is hosted in the [dcos-kafka-service](https://github.com/mesosphere/dcos-kafka-service/blob/master/images/Dockerfile) repo
```

# Configuration 

### Resources

By default, KUDO Kafka resource configuration is set to the minimum recommended values for production usage. 
But users can and should tune the configurations based on the workload requirements of their Kafka cluster.  

> ##### Tuning the resources for the Kafka Cluster

Reviewer: ```suggestion
##### Deploying Kafka with Parameters
```
